samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -121,12 +128,33 @@ public void onProcessorChange(List<String> processors) {
         ScheduleAfterDebounceTime.DEBOUNCE_TIME_MS, () -> doOnProcessorChange(processors));
   }
 
-  public void doOnProcessorChange(List<String> processors) {
+  void doOnProcessorChange(List<String> processors) {
     // if list of processors is empty - it means we are called from 'onBecomeLeader'
-    generateNewJobModel(processors);
-    if (coordinatorListener != null) {
-      coordinatorListener.onJobModelExpired();
+    List<String> currentProcessorIds = getActualProcessorIds(processors);
```

Comment:
There is also another issue of currentProcessorIds not being unique. Both the issues are not addressed yet. I would prefer to tackle them in a separate PR. Thanks for bringing it up. I will create a JIRA to track it.